aspnet/dnx/commands.rst

```diff
@@ -1,8 +1,232 @@
+The [arguments] is the name of the command to uninstall. For example::
+
+	dnx commands uninstall Microsoft.Framework.SecretManager
+	
+For additional details about the uninstall command, enter ``dnu commands uninstall -?`` from the command prompt.	
+
+Built-in global commands
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following built-in global commands are available: 
+
+	1. user-secrets
+	2. sqlservercache
+
+These commands have specific NuGet packages that must be installed. Once a global command package is installed, you can install the command using the DNU. 
+
+Building and publishing global command
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can use the .NET Development Utility (DNU) to build, package and publish a global command. A global command is contained as a console app project. Building a project produces the binary outputs for the project. Packaging produces a NuGet package that can be uploaded to a package feed (such as http://nuget.org)) and then consumed. Publishing collects all required runtime artifacts (the required DNX and packages) into a single folder so that it can be deployed as an application.
+
+When you generate a console app using the console app template, it includes a *program.cs* file containing a ``Main`` entry point to the app. After you create a console app, you can build and run the app by issuing the following DNX command::
+
+	dnx run
+
+In the console app, the *project.json* file contains the ``run`` command in the ``commands`` section. The ``dnx`` command is used to execute a managed entry point (a ``Program.Main`` function) in the assembly. When you issue the above ``dnx run`` command, DNX finds the command based on the name used for the project, then finds the ``Main`` entry point that you see in the *program.cs* file. 
+	
+For details about creating a console app with DNX, see :doc:`Creating a Cross-Platform Console App with DNX </dnx/console>`.
+	
+.. note:: The ``dnx run`` command is a shorthand for executing the entry point in the current project. It is equivalent to ``dnx [project_name]``. 
+
+When you are ready to build your console app containing your global command, use the following command to produce assemblies for the project in the given directory::
+
+	dnu build
+	
+Once the console app has been built, you can package it using the following command to create NuGet packages for the project in the given directory::
+
+	dnu pack
+	
+To publish the NuGet packages you can use the following command::
+
+	dnu publish
+	
+The ``publish`` command will package your application into a self-contained directory that can be launched. It will create the following directory structure:
+
+	- output/
+	- output/packages
+	- output/appName
+	- output/commandName.cmd
+
+The packages directory contains all the packages your command needs to run. The *appName* directory will contain all of your applications code. If you have project references, they will appear as their own directory with code at this level as well. 
```

Comment:
There are 3 commands that we skip for global install `run`, `test` and `web`. So, you can build a nuget package with those commands but they cannot be installed globally. So, for the default console application template, you have to rename the `run` command to something else like `my-cmd` if you want to make it globally installable.

Also, we prevent the following command names: `dnx`, `dnvm`, `nuget`, `dnu`. You'll get a build error if you use those names.